---
# Success: TRUE if frame is successfully delivered.
bool success